bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AbstractResourceBundleProvider.java

```diff
@@ -486,24 +487,21 @@ protected Bundle getBundle(String uid) {
     @SuppressWarnings("unchecked")
     protected void addNewProvidedObjects(List<String> newPortfolio, List<String> previousPortfolio,
             Set<E> parsedObjects) {
+        List<ProviderChangeListener<E>> snapshot = null;
+        synchronized (listeners) {
+            snapshot = new LinkedList<ProviderChangeListener<E>>(listeners);
+        }
         for (E parsedObject : parsedObjects) {
             String uid = getUID(parsedObject);
-            if (providedObjectsHolder.get(uid) == null) {
-                if (checkExistence(uid)) {
-                    continue;
-                }
-            } else if (previousPortfolio == null || !previousPortfolio.contains(uid)) {
-                logger.error("{} with UID \"{}\" already exists! Failed to create a second with the same UID!",
-                        parsedObject.getClass().getName(), uid, new IllegalArgumentException());
+            E oldelement = providedObjectsHolder.get(uid);
+            if (oldelement != null && !previousPortfolio.contains(uid)) {
+                logger.warn(
```

Comment:
Is there a special reason why you use `String.format` and not simply use parameterized logging?